web/components/liquidity-panel.tsx

```diff
@@ -0,0 +1,199 @@
+import clsx from 'clsx'
+import { useEffect, useState } from 'react'
+
+import { CPMMContract } from 'common/contract'
+import { formatMoney } from 'common/util/format'
+import { useUser } from 'web/hooks/use-user'
+import { addLiquidity, withdrawLiquidity } from 'web/lib/firebase/fn-call'
+import { AmountInput } from './amount-input'
+import { Row } from './layout/row'
+import { useUserLiquidity } from 'web/hooks/use-liquidity'
+import { Tabs } from './layout/tabs'
+import { NoLabel, YesLabel } from './outcome-label'
+import { Col } from './layout/col'
+
+export function LiquidityPanel(props: { contract: CPMMContract }) {
+  const { contract } = props
+
+  const user = useUser()
+  const lpShares = useUserLiquidity(contract, user?.id ?? '')
+
+  const [showWithdrawal, setShowWithdrawal] = useState(false)
+
+  useEffect(() => {
+    if (!showWithdrawal && lpShares && lpShares.YES && lpShares.NO)
+      setShowWithdrawal(true)
+  }, [showWithdrawal, lpShares])
+
+  return (
+    <Tabs
+      tabs={[
+        {
+          title: 'Add liquidity',
+          content: <AddLiquidityPanel contract={contract} />,
+        },
+        ...(showWithdrawal
+          ? [
+              {
+                title: 'Withdraw liquidity',
+                content: (
+                  <WithdrawLiquidityPanel
+                    contract={contract}
+                    lpShares={lpShares as any}
```

Comment:
At least cast to { YES: number, NO: number }

Reply:
ok